src/libs/Pusher/pusher.js

```diff
@@ -194,13 +201,17 @@ function subscribe(channelName, eventName, eventCallback = () => {}, isChunked =
 
         if (!channel || !channel.subscribed) {
             channel = socket.subscribe(channelName);
+            let isBound = false;
             channel.bind('pusher:subscription_succeeded', () => {
-                bindEventToChannel(channel, eventName, eventCallback, isChunked);
+                // check so that we do not bind another event with each reconnect attempt
+                if (!isBound) {
+                    bindEventToChannel(channel, eventName, eventCallback, isChunked);
+                    resolve();
+                }
+                isBound = true;
 
-                // Remove this event subscriber so we do not bind another
-                // event with each reconnect attempt
-                channel.unbind('pusher:subscription_succeeded');
-                resolve();
+                // we let this fire for the first time as well as user could wanted to fire something on success
```

Comment:
This one too. Also, I'm not super clear on what this is saying. Would you mind re-phrasing?

Reply:
Doing asap. I was thinking the same. 😸

Reply:
**Suggestion:**

>When subscribing for the first time we can register a success callback that will be called when the subscription succeeds again in the future e.g. as a result of Pusher disconnecting and reconnecting.

Reply:
Perfect suggestion.